samples/subsys/ipc/ipc_service/src/main.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <drivers/ipm.h>
+#include <sys/printk.h>
+#include <device.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <init.h>
+
+#include <ipc/ipc_service.h>
+
+#define APP_TASK_STACK_SIZE (1024)
+K_THREAD_STACK_DEFINE(thread_stack, APP_TASK_STACK_SIZE);
+static struct k_thread thread_data;
+
+static volatile unsigned int received_data;
+
+static K_SEM_DEFINE(data_rx_sem, 0, 1);
+static K_SEM_DEFINE(ept_bound_sem, 0, 1);
+
+static struct ipc_ept *ept;
+
+static int ept_bound(void *priv)
+{
+	k_sem_give(&ept_bound_sem);
+	return 0;
+}
+
+static int ept_recv(void *data, size_t len, void *priv)
+{
+	received_data = *((unsigned int *) data);
+	k_sem_give(&data_rx_sem);
+	return 0;
+}
+
+static int ept_error(const char *message, void *priv)
+{
+	printk("Endpoit error: %s", message);
+	return 0;
+}
+
+static unsigned int receive_message(void)
+{
+	k_sem_take(&data_rx_sem, K_FOREVER);
+	return received_data;
+}
+
+static int send_message(unsigned int message)
+{
+	return ipc_service_send(ept, &message, sizeof(message));
+}
+
+void app_task(void *arg1, void *arg2, void *arg3)
+{
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+	int status = 0;
+	unsigned int message = 0U;
+
+	printk("\r\nIPC Service [master] demo started\r\n");
+
+	static struct ipc_ept_cfg ept_cfg = {
+		.name = "ep",
+		.prio = -1,
+		.priv = NULL,
+		.cb = {
+			.bound    = ept_bound,
+			.received = ept_recv,
+			.error    = ept_error,
+		},
+	};
+
+	status = ipc_service_register_endpoint(&ept, &ept_cfg);
+
+	if (status < 0) {
+		printk("ipc_service_register_endpoint failed %d\n", status);
+		return;
+	}
+
+	k_sem_take(&ept_bound_sem, K_FOREVER);
+
+	while (message < 100) {
+		status = send_message(message);
+		if (status < 0) {
+			printk("send_message(%d) failed with status %d\n",
+			       message, status);
+			break;
+		}
+
+		message = receive_message();
+		printk("Master core received a message: %d\n", message);
+
+		message++;
+	}
+
+	printk("IPC Service demo ended.\n");
+}
+
+void main(void)
+{
+	printk("Starting application thread!\n");
+	k_thread_create(&thread_data, thread_stack, APP_TASK_STACK_SIZE,
+			(k_thread_entry_t)app_task,
+			NULL, NULL, NULL, K_PRIO_COOP(7), 0, K_NO_WAIT);
+
+#if defined(CONFIG_SOC_MPS2_AN521) || \
+	defined(CONFIG_SOC_V2M_MUSCA_B1)
+	wakeup_cpu1();
+	k_msleep(500);
+#endif /* #if defined(CONFIG_SOC_MPS2_AN521) */
```

Comment:
It looks only Nordic SoCs are supported so far. I think this code can be removed.